.circleci/config.yml

```diff
@@ -0,0 +1,157 @@
+          aws-secret-access-key: env_CONTEXT_ARTIFACT_S3_AWS_SECRET_ACCESS_KEY
+          aws-region: env_CONTEXT_ARTIFACT_S3_AWS_REGION
+          arguments: '--metadata "{\"x-amz-artifact-id\": \"${CIRCLE_TAG}\" }"'
+  upload_checksum:
+    description: "upload artifact checksum to s3"
+    steps:
+      - s3/copy:
+          from: artifact-info.txt
+          to: 's3://${CONTEXT_ARTIFACT_S3_BUCKET}/${CIRCLE_PROJECT_REPONAME}/'
+          aws-access-key-id: env_CONTEXT_ARTIFACT_S3_AWS_ACCESS_KEY_ID
+          aws-secret-access-key: env_CONTEXT_ARTIFACT_S3_AWS_SECRET_ACCESS_KEY
+          aws-region: env_CONTEXT_ARTIFACT_S3_AWS_REGION
+  docker_build:
+    description: "Build Docker image"
+    steps:
+      - setup_remote_docker
+      - run: 
+          name: Building docker image
+          command: |
+            docker build -t ${DOCKHUB_ORGANISATION}/deriv-com:${CIRCLE_TAG:-latest} .
+  docker_push:
+    description: "Push image to docker hub"
+    steps:
+      - run: 
+          name: Pushing Image to docker hub
+          command: |
+            echo $DOCKERHUB_PASSWORD | docker login -u $DOCKERHUB_USERNAME --password-stdin
+            docker push ${DOCKHUB_ORGANISATION}/deriv-com:${CIRCLE_TAG:-latest}
+  k8s_deploy:
+    description: "Deploy to k8s cluster"
+    parameters:
+      k8s_svc_name:
+        type: string
+        default: "staging-deriv-com"
+    steps:
+      - k8s/install-kubectl
+      - run: 
+          name: Deploying to k8s cluster for service << parameters.k8s_svc_name >>
+          command: |
+            echo $CA_CRT | base64 --decode > ca.crt
+            kubectl --server=${KUBE_SERVER} --certificate-authority=ca.crt --token=$SERVICEACCOUNT_TOKEN set image deployment/<< parameters.k8s_svc_name >> << parameters.k8s_svc_name >>=${DOCKHUB_ORGANISATION}/deriv-com:${CIRCLE_TAG}
+jobs:
+  test:
+    docker:
+      - image: circleci/node:lts-stretch
+        environment:
+          - GATSBY_CPU_COUNT=2
+    steps:
+      - git_checkout_from_cache
+      - npm_install_from_cache
+      - npm_test
+      - build
+  release:
+    docker:
+      - image: circleci/node:lts-stretch
+        environment:
+          - GATSBY_CPU_COUNT=2
+    steps:
+      - git_checkout_from_cache
+      - npm_install_from_cache
+      - build
+      - compress
+      - upload_artifact  # uploading the built code to s3 to create a backup of key services separate from Kubernetes deployment
+      - upload_checksum  # uploading compressed artifact checksum to cross match artifact fingerprint
+      - docker_build
+      - docker_push
+      - k8s_deploy
+workflows:
+  build_and_push_dev:
+    jobs:
+      - test:
+          filters:
+            branches:
+              only: /^dev/
+  build_and_push_master:
+    jobs:
+      - release:
+          filters:
+            branches:
+              only: /^master/
+          context: binary-frontend-artifact-upload  # using context to retrieve shared secret from circle ci
```

Comment:
Can we add staging release to merge with `dev` too?

Reply:
yeah, sure. i will be writing the pipeline as well